core/src/main/java/com/graphhopper/storage/index/LocationIndexTree.java

```diff
@@ -564,6 +572,133 @@ protected boolean check(int node, double normedDist, int wayIndex, EdgeIteratorS
         return closestMatch;
     }
 
+    /**
+     * Returns all edges that are within the specified radius around the queried position.
+     *
+     * @param queryLat latitude to search from
+     * @param queryLon longitude to search from
+     * @param edgeFilter only include edges from edgeFilter
+     * @param radius include all results within this radius (in meters) - assuming it's not
+     *        bounded by maxIterations.
+     * @param maxIterations if not null, search only a square of (2 * maxIterations - 1) tiles
+     *        wide/high (with center tile containing (queryLat, queryLon)). Note that it is
+     *        possible that the search radius exceeds this square - in this case, an exception
+     *        will be thrown.
+     */
+    public List<QueryResult> findWithinRadius(final double queryLat, final double queryLon,
+            final EdgeFilter edgeFilter, final double radius, final Integer maxIterations) {
+
+        // check the max iterations:
+        int requiredIterationsForThisRadius = (int) Math.ceil(radius / this.minResolutionInMeter)
+                + 1; // note that this is a conservative upper bound
+        int _maxIterations;
+        if (maxIterations == null) {
+            // used the maximum iterations required:
+            _maxIterations = requiredIterationsForThisRadius;
+        } else {
+            // check the provided maxIterations will include this radius:
+            if (maxIterations < requiredIterationsForThisRadius)
+                throw new IllegalArgumentException("A minimum of " + requiredIterationsForThisRadius
+                        + " iterations are required for a radius of " + radius + "m, but a "
+                        + "maxIterations of " + maxIterations + " was provided. Behaviour is "
+                        + "undefined in this case, so please either decrease maxIterations to "
+                         + requiredIterationsForThisRadius + ", or decrease the radius.");
+            _maxIterations = maxIterations;
+        }
+        
+        final double returnAllResultsWithin = distCalc.calcNormalizedDist(radius);
+        // implement a cheap priority queue via List, sublist and Collections.sort
+        final List<QueryResult> queryResults = new ArrayList<QueryResult>();
+        GHIntHashSet set = new GHIntHashSet();
```

Comment:
What does the set contain? Needed check the code again to see where it was passed.